c10/util/SymIntArrayRef.h

```diff
@@ -0,0 +1,190 @@
+// This file defines `SymIntArrayRef` which serves as the view onto std::vector<SymInt>
```

Comment:
You've got a bit of a dependency problem here, as you are including a file from ATen/core here that lives in c10/util. That is not kosher. I think the correct resolution is to move this header to ATen/core, and then remove the `sym_sizes` method from `TensorImpl.h`. We'll implement the sym sizes extraction separately, probably on `Tensor`, but there's no point in doing it now since we can't actually but symbolic sizes inside tensor yet.